zeus/device/cpu/rapl.py

```diff
@@ -0,0 +1,180 @@
+
+"""
+
+from __future__ import annotations
+
+import os
+import contextlib
+from typing import Sequence
+
+import zeus.device.cpu.common as cpu_common
+from zeus.device.cpu.common import CpuDramMeasurement
+from zeus.device.exception import ZeusBaseCPUError
+from zeus.utils.logging import get_logger
+
+logger = get_logger(name=__name__)
+
+RAPL_DIR = "/sys/class/powercap/intel-rapl"
+
+
+def rapl_is_available() -> bool:
+    """Check if RAPL is available."""
+    if not os.path.exists(RAPL_DIR):
+        logger.info("RAPL is not supported on this CPU.")
+        return False
+    logger.info("RAPL is available.")
+    return True
+
+
+class ZeusRAPLNotSupportedError(ZeusBaseCPUError):
+    """Zeus CPU exception class Wrapper for RAPL not supported on CPU."""
+
+    def __init__(self, message: str) -> None:
+        """Initialize Zeus Exception."""
+        super().__init__(message)
+
+
+class ZeusRAPLFileInitError(ZeusBaseCPUError):
+    """Zeus CPU exception class Wrapper for RAPL file initialization error on CPU."""
+
+    def __init__(self, message: str) -> None:
+        """Initialize Zeus Exception."""
+        super().__init__(message)
+
+
+class RAPLFile:
+    """RAPL File class for each RAPL file.
+
+    This class defines the interface for interacting with a RAPL file for a package. A package can
+    be a CPU or DRAM
+    """
+
+    def __init__(self, path: str) -> None:
+        """Initialize RAPL file object. Each RAPL file object manages one energy_uj file."""
+        self.path = path
+        self.energy_uj_path = os.path.join(path, "energy_uj")
+        try:
+            with open(os.path.join(path, "name"), "r") as name_file:
+                self.name: str = name_file.read().strip()
+        except FileNotFoundError as err:
+            raise ZeusRAPLFileInitError("Error reading package name") from err
+        try:
+            with open(self.energy_uj_path) as energy_file:
+                self.last_energy = int(float(energy_file.read().strip()) / 10**3)
+        except FileNotFoundError as err:
+            raise ZeusRAPLFileInitError("Error reading package energy") from err
+        try:
+            with open(
+                os.path.join(path, "max_energy_range_uj"), "r"
+            ) as max_energy_file:
+                self.max_energy_range_uj = int(
+                    float(max_energy_file.read().strip()) / 10**3
+                )
+        except FileNotFoundError as err:
+            raise ZeusRAPLFileInitError(
+                "Error reading package max energy range"
+            ) from err
+
+    def __str__(self) -> str:
+        """Return a string representation of the RAPL file object."""
+        return f"Path: {self.path}\nEnergy_uj_path: {self.energy_uj_path}\nName: {self.name}\
+        \nLast_energy: {self.last_energy}\nMax_energy: {self.max_energy_range_uj}"
+
+    def _read(self) -> int:
```

Comment:
I see that code outside this class is calling this method. Then this method is actually public and should be `def read(self) -> int`.  If `_read_delta` ends up being called somewhere else as well, the same applies.